src/grid/grid.component.ts

```diff
@@ -482,6 +511,7 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.calcWidth = this._width && this._width.indexOf("%") === -1 ? parseInt(this._width, 10) : 0;
         this.calcHeight = 0;
         this.calcRowCheckboxWidth = 0;
+        this.rowHeight = this.rowHeight ? this.rowHeight : this.defaultRowHeight;
```

Comment:
In this case the `rowHeight` overrides the `displayDensity` option. Maybe we should actually make the `rowHeight` private and control the height only through the `displayDensity` internally? This way we would give less control, but be sure that the grid looks optimal.

Reply:
I believe rowHeight should remain public, regardless of the displayDensity. When you set rowHeight explicitly as a user you change the height of the row and knowingly stray from the recommended default value we set. The displayDensity controls the cell padding as well.